### PR TITLE
Update the lint metrics to match agains the rule URL

### PR DESCRIPTION
**Pull request by daghack**
Update the lint metrics to match agains the rule URL rather than a prefix on the lint rule.
Required to support  https://github.com/moby/buildkit/pull/5012

### CODE REVIEW

File: util/progress/metricwriter.go [hunk 1]
```diff
@@ -446,14 +448,22 @@ func newLintMetricRecorder(meter metric.Meter, attrs attribute.Set) *lintMetricR
...
 func (mr *lintMetricRecorder) Record(ss *client.SolveStatus) {
 	for _, warning := range ss.Warnings {
-		m := reLintMessage.FindSubmatch(warning.Short)
+		m := reLintMessage.FindSubmatch([]byte(warning.URL))
 		if m == nil {
 			continue
 		}
 
-		ruleName := string(m[1])
+		ruleName := kebabToCamel(string(m[1]))
```
Comment (tonistiigi) [inline util/progress/metricwriter.go:466]: This should need a length check (already in old code)

**Review by tonistiigi — CHANGES_REQUESTED** [review]

File: util/progress/metricwriter.go (hunk 1 above)
Comment (daghack) [reply util/progress/metricwriter.go:466]: Do you mean a length check on the result from `FindSubmatch`?

File: util/progress/metricwriter.go [hunk 2]
```diff
@@ -446,14 +448,22 @@ func newLintMetricRecorder(meter metric.Meter, attrs attribute.Set) *lintMetricR
...
+	}
+	return strings.Join(words, "")
+}
+
 func (mr *lintMetricRecorder) Record(ss *client.SolveStatus) {
 	for _, warning := range ss.Warnings {
-		m := reLintMessage.FindSubmatch(warning.Short)
-		if m == nil {
+		m := reLintMessage.FindSubmatch([]byte(warning.URL))
+		if m == nil || len(m) < 2 {
```
Comment (tonistiigi) [inline util/progress/metricwriter.go:462]: nit: technically `nil` check is not needed anymore

File: util/progress/metricwriter.go (hunk 1 above)
Comment (tonistiigi) [reply util/progress/metricwriter.go:466]: Yes, to avoid panic if `[]` is returned.

File: util/progress/metricwriter.go (hunk 2 above)
Comment (daghack) [reply util/progress/metricwriter.go]: Removed redundant nil!

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**